src/main.cpp

```diff
@@ -3087,8 +3087,14 @@ bool ConnectBlock(const CBlock& block, CValidationState& state, CBlockIndex* pin
         //   to ConnectBlock() (and thus never on the main chain) will stay with
         //   these set to null.
         if (chainparams.GetConsensus().NetworkUpgradeActive(pindex->nHeight, Consensus::UPGRADE_NU5)) {
+            try {
+                pindex->hashFinalOrchardRoot = orchard_tree.root();
+            } catch (const OrchardAnchorBottom& e) {
+                return state.DoS(100,
+                    error("ConnectBlock(): derived the bottom value computing the root of the Orchard commitment tree"),
+                    REJECT_INVALID, "bad-orchard-root-in-block");
+            }
```

Comment:
This check needs to be performed immediately after the transaction loop, because the `view.PushAnchor(orchard_tree)` call above here will call `tree.root()` and trigger the exception before we reach this point.